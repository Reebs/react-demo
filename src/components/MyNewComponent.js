/**
 * Created by dbgreenfield on 10/16/17.
 */
import React, { Component } from 'react';
import Banner from './Banner'

export default class MyNewComponent extends Component {

    constructor(props){
        super(props)
        this.fieldchange = this.fieldChange.bind(this)
        this.state = {"fieldValue": this.props.initValue,"banner":""}
        this.onclick = this.onClick.bind(this)
    }

    fieldChange(event){
        this.setState({"fieldValue": event.target.value})
    }

    onClick(event){
        this.setState({banner:this.state.fieldValue})
    }

    render(){
        
        return <div>

            <Banner banner={this.state.banner}/>
            <input onChange={this.fieldchange}/>
            {' '}<button onClick={this.onclick}>click to change banner</button>
            <p>{this.state.fieldValue}</p>


        </div>
    }
}
