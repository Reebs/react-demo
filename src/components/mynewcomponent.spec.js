import React from 'react';
import renderer from 'react-test-renderer'
import MyNewComponent from "./MyNewComponent";

test('MyNewComponent  shows up when rendered', () => {
  
  const component = renderer.create(<MyNewComponent  initValue='hello world!' />)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
