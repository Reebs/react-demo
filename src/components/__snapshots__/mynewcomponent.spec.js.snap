// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyNewComponent  shows up when rendered 1`] = `
<div>
  <h1>
    
  </h1>
  <input
    onChange={[Function]}
  />
   
  <button
    onClick={[Function]}
  >
    click to change banner
  </button>
  <p>
    hello world!
  </p>
</div>
`;
